### PR TITLE
[NEW] Adds Collapsible Channel Lists

### PR DESCRIPTION
**Pull request by MartinSchoeler**
@RocketChat/core 

Continuation of #6316 
This adds the ability to collapse channels lists, like this
screenshot from 2017-04-06 17-36-39
screenshot from 2017-04-06 17-43-49

This is a toggle-able user preference
screenshot from 2017-04-06 17-35-48

### CODE REVIEW

File: packages/rocketchat-i18n/i18n/en.i18n.json [hunk 1]
```diff
@@ -329,6 +329,7 @@
   "Closed": "Closed",
   "Closed_by_visitor": "Closed by visitor",
   "Closing_chat": "Closing chat",
+  "Collapse_Channels":"Collapsible Channel Lists",
```
Comment (rodrigok) [inline packages/rocketchat-i18n/i18n/en.i18n.json]: Please keep the pattern, add a space after `:`

File: packages/rocketchat-ui-sidenav/client/sideNav.js [hunk 2]
```diff
@@ -50,6 +50,10 @@ Template.sideNav.helpers({
 		} else {
 			return this.template;
 		}
+	},
+
+	collapseChannel() {
+		return {collapsible : Meteor.user().settings.preferences.collapseChannels};
```
Comment (rodrigok) [inline packages/rocketchat-ui-sidenav/client/sideNav.js]: Ident the object and remove the space before the colon

**Comment by rodrigok** [comment]
First timing trying to run it I got this error:
image
The code throwing the error is:
image

File: packages/rocketchat-ui-sidenav/client/sideNav.js [hunk 3]
```diff
@@ -85,3 +104,11 @@ Template.sideNav.onRendered(function() {
 
 	return Meteor.defer(() => menu.updateUnreadBars());
 });
+
+Template.sideNav.onCreated(function() {
+	this.autorun(() => {
+		if (!Meteor.user().settings.preferences.collapseChannels) {
```
Comment (geekgonecrazy) [inline packages/rocketchat-ui-sidenav/client/sideNav.js]: Don't we normally use the Rocketchat.settings to set and retrieve settings?

Comment (MartinSchoeler) [inline packages/rocketchat-ui-sidenav/client/sideNav.js]: @geekgonecrazy I only saw this approach with the user preferences. Can we use Rocketchat.settings to get the user preferences?

Comment (geekgonecrazy) [inline packages/rocketchat-ui-sidenav/client/sideNav.js]: Ah.. right I don't think we've created a way to retrieve preferences.  Nvm :)

**Comment by MartinSchoeler** [comment]
@rodrigok could you do a review again?

**Comment by rodrigok** [comment]
image

**Comment by MartinSchoeler** [comment]
closing this due the sidenav redesign

**Pull request closed without merging**